### ISSUE TITLE
Resolve event listing page discrepancies between full build and preview 

### ISSUE THREAD

**Issue by mpelzsherman** [department-of-veterans-affairs/va.gov-team#27433]
# Description

There are some discrepancies between preview and full build on the system/events page

http://preview-prod.vfs.va.gov/preview?nodeId=2807 vs
https://www.va.gov/pittsburgh-health-care/events/

Preview still shows a past event, and shows the Upcoming/Past toggle.

It looks like the issue with showing past events might be unique to Pittsburgh. Wilmington does not have this issue:

https://www.va.gov/wilmington-health-care/events/
http://preview-prod.vfs.va.gov/preview?nodeId=3037

## Acceptance Criteria
- [x] Experience is consistent between between preview and prod - **as it relates to published content only**
- [x] Published, Draft and in Review events are visible on event listing pages in Preview
- [x] Archived events are not visible on event listing pages in Preview
- [x] Display the upcoming/past event toggle consistently 
- [x] Past events don't display as a spotlight 
- [x] Spotlight events (Published, Draft and in Review) must be current or upcoming and only display on upcoming events page 
- [x] Navigation between pages is not possible in Preview

**Comment by mpelzsherman** [comment]
Re: "Published, Draft and in Review events are visible on event listing pages in Preview"

Currently, there's a filter on the GraphQL query that restricts the results to `status = 1` (Published) events.

If we remove this filter in Preview mode (which is easy to do), I don't know how to prevent **Archived** events from showing up in Preview mode.

Is it OK to show archived events in preview? If so, please update the AC accordingly.

If we MUST omit Archived events in preview mode, I suggest we break out this requirement into its own ticket, otherwise this ticket will be blocked until the CMS team provides this information.

**Comment by mpelzsherman** [comment]
The CMS team informed me about the `moderationState` field, which can be used to filter out Archived events on the front end. They are also adding a Drupal patch to enable filtering on this field.

However, today the `moderationState` field seems to have disappeared from Drupal production. 

So that aspect of the ticket is now blocked. I'll keep working on completing the other AC's.

**Comment by mpelzsherman** [comment]
The Drupal patch caused the moderationState field to become unavailable in production, so they are rolling it back. We won't be able to filter in graphql but we can at least filter on the front end once the rollback is done.

**Comment by mpelzsherman** [comment]
Update from @swirtSJW:

> As of the end of last week, we were not able to revert the patch and have moderationState re-appear.   Neil, Nate and I put a lot of time into it but without success.   Neil created a ticket related to it and I believe it is slated for next sprint.  We just could not bump currrent sprint priorities to pour into this rabbit hole.

So the current status in prod is we can filter on `moderation_state`, but we can’t query on it.

I've re-started the ticket and will apply the filter to the GraphQL query, but remove the use of the `moderationState` field until it can be accessed.

**Comment by mpelzsherman** [comment]
Closing - validated in prod.
